app/internal/page/attorney/check_your_name.go

```diff
@@ -51,29 +52,33 @@ func CheckYourName(tmpl template.Template, lpaStore LpaStore, notifyClient Notif
 			data.Errors = data.Form.Validate()
 
 			if len(data.Errors) == 0 {
-				if data.Form.CorrectedName != "" {
-					attorneyProvidedDetails.DeclaredFullName = data.Form.CorrectedName
-					if appData.IsReplacementAttorney {
-						lpa.ReplacementAttorneyProvidedDetails.Put(attorneyProvidedDetails)
-					} else {
-						lpa.AttorneyProvidedDetails.Put(attorneyProvidedDetails)
-					}
-
-					if err := lpaStore.Put(r.Context(), lpa); err != nil {
-						return err
-					}
+				previousCorrectedName := attorneyProvidedDetails.CorrectedName
+				attorneyProvidedDetails.IsNameCorrect = data.Form.IsNameCorrect
```

Comment:
I noticed this page wasn't keeping the state of the radio buttons, and also did something simple to prevent it notifying multiple times when the name hasn't been changed.